js/utils.js

```diff
@@ -0,0 +1,32 @@
+export async function makeApiCall(
+  url,
+  method = 'get',
+  body = null,
+  credentials = 'include',
+  headers = { 'content-type': 'application/json' },
+  options = null,
+) {
+  try {
+    const response = await fetch(url, {
+      method,
+      body,
+      headers,
+      credentials,
+      ...options,
+    });
+    const res = {
+      data: await response.json(),
+      status: response.status,
+    };
+    return res;
+  } catch (err) {
+    throw err;
+  }
+}
+
+export function createElement({ type, classList = [], id }) {
```

Comment:
What if id is empty or undefined?

Reply:
You might need a utility function like this for `helpers/utils` functions, as they are globally used.

`Snippet`

```jsx
/**
 * @param value:
 * @returns boolean which returns
 * - `true` if value is empty or falsy
 * - `false`  if value is not empty or truthy
 */
function isEmpty(valueToCheck) {
  switch (typeof valueToCheck) {
    case "undefined":
      return true;
    case "string":
      return valueToCheck.trim().length === 0 || valueToCheck.length === 0;
    case "object":
      if (valueToCheck === null) {
        return true;
      } else if (Array.isArray(valueToCheck)) {
        return valueToCheck.length === 0;
      } else {
        return Object.keys(valueToCheck).length === 0;
      }
    case "number":
      return Number.isNaN(valueToCheck);
    default:
      return false;
  }
}
```